src/ga/operators/selection.rs

```diff
@@ -23,3 +23,15 @@ pub fn roulette_wheel<T: Chromosome, S: ChromosomeWrapper<T>>(population: &Vec<S
 
 	selected
 }
+
+pub fn random<T: Chromosome, S: ChromosomeWrapper<T>>(population: &Vec<S>, count: usize) -> Vec<&S> {
+	// We must use index API, as we want to return vector of references, not vector of actual items
+	let indices = rand::seq::index::sample(&mut rand::thread_rng(), population.len(), count);
+	let mut selected: Vec<&S> = Vec::with_capacity(count);
```

Comment:
We could really consider using custom, thread local memory allocator, so that we don't have to reach global memory pool every time such function is called.